src/test/regress/specs/isolation_copy_placement_vs_modification.spec

```diff
@@ -0,0 +1,106 @@
+# we use 5 as the partition key value through out the test
+# so setting the corresponding shard here is useful
+setup
+{	
+	SET citus.shard_count TO 2;
+	SET citus.shard_replication_factor TO 2;
+	CREATE TABLE test_table (x int, y int);
+	SELECT create_distributed_table('test_table', 'x');
+	
+	SELECT get_shard_id_for_distribution_column('test_table', 5) INTO selected_shard;
+}
+
+teardown
+{
+	DROP TABLE test_table;
+	DROP TABLE selected_shard;
+}
+
+session "s1"
+
+step "s1-begin"
+{
+    BEGIN;
+}
+
+# since test_table has rep > 1 simple select query doesn't hit all placements
+# hence not all placements are cached 
+step "s1-load-cache"
+{
+	TRUNCATE test_table;
+}
+
+step "s1-insert"
+{
+	INSERT INTO test_table VALUES (5, 10);
+}
+
+step "s1-update"
+{
+	UPDATE test_table SET y = 5 WHERE x = 5;
+}
+
+step "s1-delete"
+{
+	UPDATE test_table SET y = 5 WHERE x = 5;
+}
+
+step "s1-select"
+{
+	SELECT count(*) FROM test_table WHERE x = 5;
+}
+
+step "s1-copy"
+{
+	COPY test_table FROM PROGRAM 'echo "1,1\n2,2\n3,3\n4,4\n5,5"' WITH CSV;
+}
+
+step "s1-commit"
+{
+	COMMIT;
+}
+
+session "s2"
+
+step "s2-begin"
+{
+	BEGIN;
+}
+
+step "s2-set-placement-inactive"
+{
+	UPDATE pg_dist_shard_placement SET shardstate = 3 WHERE shardid IN (SELECT * FROM selected_shard) AND nodeport = 57638;
+}
+
+step "s2-repair-placement"
+{
+	SELECT master_copy_shard_placement((SELECT * FROM selected_shard), 'localhost', 57637, 'localhost', 57638);
+}
+
+step "s2-commit"
+{
+	COMMIT;
+}
+
+step "s2-print-content"
+{
+	SELECT 
+		nodeport, success, result 
+	FROM 
+		run_command_on_placements('test_table', 'select y from %s WHERE x = 5')
+	WHERE
+		shardid IN (SELECT * FROM selected_shard)
+	ORDER BY
+		nodeport;
+}
+
+permutation "s1-load-cache" "s1-insert" "s1-begin" "s1-select" "s2-set-placement-inactive" "s2-begin" "s2-repair-placement" "s1-update" "s2-commit" "s1-commit" "s2-print-content"
```

Comment:
Brief one-line explanations of the tests would be nice, since it's not very readable, e.g.

```
# repair a placement while concurrently performing an update
```